pandas/core/indexes/base.py

```diff
@@ -3068,13 +3068,23 @@ def get_value(self, series, key):
         # if we have something that is Index-like, then
         # use this, e.g. DatetimeIndex
         s = getattr(series, '_values', None)
-        if isinstance(s, (ExtensionArray, Index)) and is_scalar(key):
-            try:
-                return s[key]
```

Comment:
if you just use ``.get_loc(key)`` on an Index as well does this work? (perf should be similar). That way we can avoid separating this.